### PR TITLE
feat(table): Add <Table> component and wire up users

### PR DESCRIPTION
**Pull request by damassi**
This adds a new `<Table>` component and wires it up to the /users route.

For this I used [react-table](https://react-table.tanstack.com/) which is a cool table building toolchain, rather than a table component, which allows us to keep full control of everything including styles. It does a lot and I think it will scale out well to all of our needs, whatever they may be. 

table

The wrapper `<Table />` component that I wrote around it looks like so:

```tsx
 <Table
  columns={[
    {
      Header: "Name",
      accessor: "name", // accessor is the "key" in the data
    },
    {
      Header: "Email",
      accessor: "email",
    },
    {
      Header: "ID",
      accessor: "internalID",
    },
  ]}
  data={users}
  onRowClick={(row) => {
    router.push(`/users/${row.values.internalID}`)
  }}
/>
```

As things expand I imagine this API will grow.

### CODE REVIEW

File: src/components/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { Box, Text, THEME_V3 } from "@artsy/palette"
+import { useMemo } from "react"
+import { Row, useTable } from "react-table"
+import styled from "styled-components"
+
+interface TableProps {
+  columns: any[]
```
Comment (damassi) [inline src/components/Table.tsx:7]: Need to figure out generic types

File: src/components/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  data: any[]
+  onRowClick: (row: Row) => void
+}
+
+export const Table: React.FC<TableProps> = ({ columns, data, onRowClick }) => {
+  const columnsMemo = useMemo(() => columns, [columns])
+  const dataMemo = useMemo(() => data, [data])
```
Comment (dzucconi) [inline src/components/Table.tsx:14]: Why do you have to memo these props?

File: src/components/Table.tsx (hunk 1 above)
Comment (dzucconi) [reply src/components/Table.tsx:7]: Something like:

```tsx
interface TableProps<T, U> {
  columns: T[]
  data: U[]
}

export const Table = <T, U>({ columns, data }: TableProps<T, U>) => {
  // ...
```

**Review by dzucconi — APPROVED** [review]

File: src/components/Table.tsx (hunk 2 above)
Comment (damassi) [reply src/components/Table.tsx:14]: This is what the docs said was required

File: src/components/Table.tsx (hunk 1 above)
Comment (damassi) [reply src/components/Table.tsx:7]: 👍 will address next week

**Pull request merged**